nats/src/connector.rs

```diff
@@ -47,53 +46,79 @@ pub(crate) struct Connector {
     tls_config: Arc<ClientConfig>,
 }
 
-impl Connector {
-    /// Creates a new connector with the URLs and options.
-    pub(crate) fn new(urls: Vec<ServerAddress>, options: Arc<Options>) -> io::Result<Connector> {
-        let mut tls_config = options.tls_client_config.clone();
-
-        // Include system root certificates.
-        //
-        // On Windows, some certificates cannot be loaded by rustls
-        // for whatever reason, so we simply skip them.
-        // See https://github.com/ctz/rustls-native-certs/issues/5
-        let roots = match rustls_native_certs::load_native_certs() {
-            Ok(store) | Err((Some(store), _)) => store.roots,
-            Err((None, _)) => Vec::new(),
-        };
-        for root in roots {
-            tls_config.root_store.roots.push(root);
-        }
+fn configure_tls(options: &Arc<Options>) -> Result<ClientConfig, Error> {
+    let mut root_store = rustls::RootCertStore::empty();
+
```

Comment:
Add a comment explaining that if condition

from async:
`    // load native system certs only if user did not specify them.`